### PR TITLE
✨ vote: get voting power from velodrome and extra

### PR DESCRIPTION
**Pull request by cruzdanilo**

### CODE REVIEW

File: contracts/periphery/VotePreviewer.sol [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    exa = exa_;
+    pool = pool_;
+    gauge = gauge_;
+    extraLending = extraLending_;
+    extraReserveId = extraReserveId_;
+  }
+
+  function externalVotes(address account) external view returns (uint256 votes) {
+    uint256 liquidity = pool.balanceOf(account) + gauge.balanceOf(account);
+    votes += liquidity.mulDivDown(exa == pool.token0() ? pool.reserve0() : pool.reserve1(), pool.totalSupply());
```
Comment (sebipap) [inline contracts/periphery/VotePreviewer.sol:35]: how does `pool.token0()` evaluate as a bool for the ternary statement? I understand that `token0` and `token1` are the 2 tokens in VELO pool, but how is it falsy?

File: contracts/periphery/VotePreviewer.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    uint256[] memory reserveIds = new uint256[](1);
+    reserveIds[0] = extraReserveId;
+    IExtraLending.PositionStatus[] memory positions = extraLending.getPositionStatus(reserveIds, account);
+    votes += positions[0].eTokenStaked + positions[0].eTokenUnStaked;
+  }
```
Comment (sebipap) [inline contracts/periphery/VotePreviewer.sol:41]: why do we count `eTokenUnStaked`? Don't we risk double counting?

Comment (cruzdanilo) [reply contracts/periphery/VotePreviewer.sol:41]: we need to count both staked and unstaked tokens. for example, the treasury tokens (used in the test) are all unstaked.

Comment (sebipap) [reply contracts/periphery/VotePreviewer.sol:41]: cool

**Review by sebipap — APPROVED** [review]

**Review by itofarina — APPROVED** [review]

**Pull request merged**